### PR TITLE
SHIPPING-1969 Add GET and POST api-specs for Pickup Methods

### PR DESCRIPTION
**Pull request by agchan12**
This is branched of https://github.com/bigcommerce/api-specs/pull/607

DON'T MERGE UNLESS https://github.com/bigcommerce/api-specs/pull/607 has been merged.

Add Open API documentation for new GET and POST pickup methods operation.

⚠️ This API is not generally available, so the documentation shouldn't be made public yet. It'll first be part of a beta.

@bigcommerce/dev-docs @bc-andreadao @bigcommerce/shipping

### CODE REVIEW

**Review by theromulans — APPROVED** [review]

File: reference/pickup.v3.yaml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            type: integer
+            format: int64
+        - name: store_hash
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: The request has been processed and a list of Pickup Methods is returned successfully.
```
Comment (bc-andreadao) [inline reference/pickup.v3.yaml:43]:
```suggestion
          description: The request has been processed and a list of Pickup Methods has been returned successfully.
```

File: reference/pickup.v3.yaml [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                      $ref: '#/components/schemas/PickupMethod'
+                  meta:
+                    $ref: '#/components/schemas/MetaCollection'
+    post:
+      tags:
+        - Pickup Methods
+      description: |-
+        Creates one or many pickup methods.
+        
+        This is a batch operation where creating up to 10 pickup methods objects can be done in one request.
```
Comment (bc-andreadao) [inline reference/pickup.v3.yaml:61]:
```suggestion
        This is a batch operation that can create up to 10 pickup methods objects in one request.
```

File: reference/pickup.v3.yaml [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                    type: object
+    put:
+      tags:
+        - Pickup Methods
+      description: |-
+        Update existing pickup methods.
+        
+        This is a batch operation where updating up to 10 pickup methods objects can be done in one request.
+        
+        Only the `id` property of the pickup method object is required as part of the request. Where a pickup method property is not provided, it will maintain its existing value.
```
Comment (bc-andreadao) [inline reference/pickup.v3.yaml:101]:
```suggestion
        Only the `id` property of the pickup method object is required as part of the request. Pickup method properties that are not provided will maintain their existing values.
```

**Review by bc-andreadao — COMMENTED** [review]
Added suggestions for grammar

**Review by bc-andreadao — APPROVED** [review]

**Pull request merged**